HPXMLtoOpenStudio/resources/vehicle.rb

```diff
@@ -0,0 +1,223 @@
+        weekday_charge, weekday_discharge = Schedule.split_signed_charging_schedule(vehicle.ev_charging_weekday_fractions)
+        weekend_charge, weekend_discharge = Schedule.split_signed_charging_schedule(vehicle.ev_charging_weekend_fractions)
+        charging_schedule = MonthWeekdayWeekendSchedule.new(model, charge_name, weekday_charge, weekend_charge, vehicle.ev_charging_monthly_multipliers, EPlus::ScheduleTypeLimitsFraction, false)
+        charging_schedule = charging_schedule.schedule
+        discharging_schedule = MonthWeekdayWeekendSchedule.new(model, discharge_name, weekday_discharge, weekend_discharge, vehicle.ev_charging_monthly_multipliers, EPlus::ScheduleTypeLimitsFraction, false)
+        discharging_schedule = discharging_schedule.schedule
+      end
+    else
+      runner.registerWarning("Both schedule file and weekday fractions provided for '#{charging_col}' and '#{discharging_col}'; weekday fractions will be ignored.") if !vehicle.ev_charging_weekday_fractions.nil?
+      runner.registerWarning("Both schedule file and weekend fractions provided for '#{charging_col}' and '#{discharging_col}'; weekend fractions will be ignored.") if !vehicle.ev_charging_weekend_fractions.nil?
+      runner.registerWarning("Both schedule file and monthly multipliers provided for '#{charging_col}' and '#{discharging_col}'; monthly multipliers will be ignored.") if !vehicle.ev_charging_monthly_multipliers.nil?
+    end
+
+    return charging_schedule, discharging_schedule
+  end
+
+  # Apply an electric vehicle to the model using the battery.rb Battery class, which assigns OpenStudio ElectricLoadCenterStorageLiIonNMCBattery and ElectricLoadCenterDistribution objects.
+  # An EMS program models the effect of ambient temperature on the effective power output.
+  # An EMS program writes a 'discharge offset' variable to omit this from aggregate home electricity outputs.
+  # If no charging/discharging schedule is provided, then the electric vehicle is not modeled.
+  # Bi-directional charging is not currently implemented
+  #
+  # @param runner [OpenStudio::Measure::OSRunner] Object typically used to display warnings
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param spaces [Hash] Map of HPXML locations => OpenStudio Space objects
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param vehicle [HPXML::Vehicle] Object that defines a single electric vehicle
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [nil]
+  def self.apply_electric_vehicle(runner, model, spaces, hpxml_bldg, vehicle, schedules_file)
+    model.getElectricEquipments.sort.each do |ee|
+      if ee.endUseSubcategory.start_with? Constants::ObjectTypeMiscElectricVehicleCharging
+        runner.registerWarning('Electric vehicle was specified as a plug load and as a battery, vehicle charging will be modeled as a plug load.')
```

Comment:
Glad to see you have a warning for this. 👍 

```suggestion
        runner.registerWarning('Electric vehicle charging was specified as both a PlugLoad and a Vehicle, the latter will be ignored.')
```
(If you accept the suggestion, there are other files that need to be updated too.)